tooling/azure-automation/resources-cleanup/src/resources_cleanup.py

```diff
@@ -0,0 +1,155 @@
+        dot_index = date_time_str.index(dot)
+        date_time_str = date_time_str[0:dot_index]
+    
+    utc_suffix = '+00:00'
+    date_time_str += utc_suffix
+    
+    return datetime.datetime.fromisoformat(date_time_str)
+
+
+def time_delta_greater_than_two_days(now: datetime.datetime, resource_group_creation_time: datetime.datetime):
+    if now == None:
+        print("now time is None")
+        return False
+    
+    if resource_group_creation_time == None:
+        # We do not want to delete the resource group if it does not have a creation timestamp.
+        print("resource_group_creation_time is None")
+        return False
+    
+    time_delta = resource_group_creation_time - now
+    n_days = abs(time_delta.days)
+
+    return n_days > 2
+
+
+def print_resources(resource_list: List[GenericResourceExpanded]):
+    for resource in resource_list:
+        print(f"- name: {resource.name}")
+        print(f"    ID: {resource.id}")
+        print(f"    type: {resource.type}")
+        print(f"    created at: {resource.created_time}")
+        print(f"    changed at: {resource.changed_time}")
+        print(f"    tags: {resource.tags}\n")
+
+
+def resource_group_has_persist_tag_as_true(resource_group: ResourceGroup):
+    if resource_group.tags == None:
+        return False
+
+    persist_tag = "persist"
+    if persist_tag not in resource_group.tags:
+        return False
+
+    return resource_group.tags[persist_tag].lower() == "true"
+
+
+def process_resource_groups_of_subscription(resource_client: ResourceManagementClient, subscription_id: str):
+    resource_groups_list = list(resource_client.resource_groups.list())
+
+    n_resource_groups = len(resource_groups_list)
+    print(f"subscription {subscription_id} has {n_resource_groups} resource groups:\n")
+
+    for resource_group in resource_groups_list:
+        process_resource_group(resource_client, resource_group)
+        print("_"*80)
+        print()
+
+
+def process_resource_group(resource_client: ResourceManagementClient, resource_group: ResourceGroup):
+    resource_group_name = resource_group.name
+        
+    resource_list = list(
+        resource_client.resources.list_by_resource_group(resource_group_name, expand = "createdTime,changedTime")
+    )
+
+    print(f"Resource group '{resource_group_name}':")
+    print(f"Tags: {resource_group.tags}\n")
+    print(f"This resource group has {len(resource_list)} resources \n")
+    
+    if VERBOSE:
+        print_resources(resource_list)
+    
+    if resource_group_has_persist_tag_as_true(resource_group):
+        print(f"Persist tag is true, this resource group should NOT be deleted, skipping.")
+        return
+
+    now = datetime.datetime.now(datetime.timezone.utc)
+    resource_group_creation_time = get_creation_time_of_resource_group(resource_group)
+    if not time_delta_greater_than_two_days(now, resource_group_creation_time):
+        print(f"This resource group should NOT be deleted, skipping.")
```

Comment:
I suppose with this print statement, we can just leave the above comment printing suggestion alone.

Reply:
Yes, this was the combination of comment + log that was more clear to me after trying some.